File: front/app/types/physical/component_slot.ts
import * as v from 'valibot'

// Components Slot
export const componentSlotSchema = v.object({
    id: v.string(),
    name: v.pipe(v.string(), v.nonEmpty()),
    description: v.nullish(v.string()),
    componentTypeId: v.nullish(v.string()),
})

export const componentSlotFormSchema = v.partial(componentSlotSchema, ['id'])

// Export des types
export type ComponentSlotType = v.InferInput<typeof componentSlotSchema>

File: api/app/graphql/app/physical/types/component_slot.rb
# frozen_string_literal: true

module App
  module Physical
    module Types
      # ComponentSlotType
      class ComponentSlot < ::Types::BaseObject
        graphql_name 'ComponentSlotType'

        field :id, ID, null: true
        field :name, String, null: false
        field :component_type_id, ID, null: true
        field :description, String, null: true
      end
    end
  end
end
